File: Vida/lib/models/user_model.dart
class UserModel {
  int? userId;
  String? username;
  bool? authentication;
  String? fullName;
  String? phoneNumber;
  UserModel({
    this.userId,
    this.username,
    this.authentication,
    this.fullName,
    this.phoneNumber,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      userId: json["userId"] as int,
      username: json["username"],
      authentication: json["authentication"] as bool,
      fullName: json["fullName"],
      phoneNumber: json["phoneNumber"],
    );
  }

  Map<String, dynamic> toJson({int depth = 0}) => {
        "userId": userId,
        "username": username,
        "authentication": authentication,
        "fullName": fullName,
        "phoneNumber": phoneNumber
      };

  @override
  String toString() => toJson().toString();
}

File: Server/src/main/java/com/pbl3/musicapplication/model/model/UserModel.java
package com.pbl3.musicapplication.model.model;

import com.pbl3.musicapplication.model.entity.User;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
@ToString
public class UserModel {
    private Integer userId;
    private String username;
    private boolean authentication;
    private String fullName;
    private String phoneNumber;

    public UserModel(User entity) {
        this.userId = entity.getUserId();
        this.username = entity.getUsername();
        this.fullName = entity.getFullName();
        this.phoneNumber = entity.getPhoneNumber();
    }

}
